apps/www/registry/default/ui/date-range-picker.tsx

```diff
@@ -0,0 +1,210 @@
+"use client"
+
+import * as React from "react"
+import { Calendar as CalendarIcon } from "lucide-react"
+import { addDays, format } from "date-fns"
+import { DateRange } from "react-day-picker"
+
+import { cn } from "@/lib/utils"
+import { Button } from "@/registry/default/ui/button"
+import { Calendar } from "@/registry/default/ui/calendar"
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/registry/default/ui/popover"
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/registry/default/ui/select"
+
+interface DateRangePickerProps extends React.HTMLAttributes<HTMLDivElement> {
+  showExternalPresets?: boolean
+  showInternalPresets?: boolean
+  numberOfMonths?: 2 | 3 | 4 | 5 | 6
+}
+
+export function DateRangePicker({
+  showExternalPresets,
+  showInternalPresets,
+  numberOfMonths = 2,
+  className,
+}: DateRangePickerProps) {
+  const [date, setDate] = React.useState<DateRange | undefined>({
+    from: new Date(2023, 0, 20),
+    to: addDays(new Date(2023, 0, 20), 20),
+  })
+
+  const handlePresetSelect = (value: string) => {
+    switch (value) {
+      case "last7Days":
+        setDate({
+          from: addDays(new Date(), -7),
+          to: new Date(),
+        })
+        break
+      case "last30Days":
+        setDate({
+          from: addDays(new Date(), -30),
+          to: new Date(),
+        })
+        break
+      case "monthToDate":
+        setDate({
+          from: new Date(new Date().getFullYear(), new Date().getMonth(), 1),
+          to: new Date(),
+        })
+        break
+      case "yearToDate":
+        setDate({
+          from: new Date(new Date().getFullYear(), 0, 1),
+          to: new Date(),
+        })
+        break
+      case "thisYear":
+        setDate({
+          from: new Date(new Date().getFullYear(), 0, 1),
+          to: new Date(new Date().getFullYear(), 11, 31),
+        })
+        break
+      case "lastYear":
+        setDate({
+          from: new Date(new Date().getFullYear() - 1, 0, 1),
+          to: new Date(new Date().getFullYear() - 1, 11, 31),
+        })
+        break
+      default:
+        break
+    }
+  }
+
+  return (
+    <div className={cn("inline-flex", className)}>
+      <Popover>
+        <PopoverTrigger asChild>
+          <Button
+            id="date"
+            variant={"outline"}
+            className={cn(
+              "w-[260px] justify-start text-left font-normal",
+              !date && "text-muted-foreground",
+              showExternalPresets && "rounded-r-none",
+            )}
+          >
+            <CalendarIcon className="mr-2 h-4 w-4" />
+            {date?.from ? (
+              date.to ? (
+                <>
+                  {format(date.from, "LLL dd, y")} -{" "}

```

Comment:
Why use space character on the left side of the dash character and then use spaced quotes on the right side?

Reply:
```suggestion
                 <>
                     {format(date.from, "LLL dd, y")} - {format(date.to, "LLL dd, y")}                
                 </>
```
Somehow my code formatter adds it unintentionally.